Use a deque for the ready_ queue so that build order is not randomized
by using a set of pointers.

Index: src/state.h
--- src/state.h.orig
+++ src/state.h
@@ -17,6 +17,7 @@
 
 #include <map>
 #include <set>
+#include <deque>
 #include <string>
 #include <vector>
 
@@ -61,7 +62,7 @@ struct Pool {
   void DelayEdge(Edge* edge);
 
   /// Pool will add zero or more edges to the ready_queue
-  void RetrieveReadyEdges(std::set<Edge*>* ready_queue);
+  void RetrieveReadyEdges(std::deque<Edge*>* ready_queue);
 
   /// Dump the Pool and its edges (useful for debugging).
   void Dump() const;
